content/kb/troubleshooting-failure-to-delete-orphaned-pod-volume-directory.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Troubleshooting: Failure to delete orphaned pod volume directories"
+author: Ray Chang
+draft: false
+date: 2023-03-23
+categories:
+  - "csi"
+---
+
+## Applicable versions
+
+All Longhorn versions.
+
+## Symptoms
+
+When a worker node crashes with some pods running on it, many error messages as shown below keep appearing in the kubelet.
+
+```
+orphaned pod <pod-uid> found, but error not a directory occurred when trying to remove the volumes dir
+```
+
+## Reason
+
+This is caused by the kubelet cleaning up volume folders but disconnecting from the CSI driver when the node crashes. Therefore, when the kubelet tries to delete `vol_data.json` to unmount the volume, it fails and causes further deletion of the old volume mount point to fail.
```

Comment:
> when the kubelet tries to delete `vol_data.json` to unmount the volume

Do you mean that when the kubelet tries to teardown a volume by sending CSI request to longhorn-csi-plugin, it fails because longhorn-csi-plugin is not running. Then kubelet cannot delete `vol_data.json` to unmount the volume as the csi call failed.

Reply:
Yes, I think it's because the connection between kubelet and longhorn-csi-plugin is disconnected due to an abnormal node, but it tries to tear down the volume, which fails to delete `vol_data.json`.